### ISSUE TITLE
iOS 13.1 - contextmenu/longtap issue on Markers

### ISSUE THREAD

**Issue by eliboni** [Leaflet/Leaflet#6865]
This is linked to #6817 fixed by  #6855.

How to reproduce: http://www.mappite.org/tap/testTap.html
Leaflet Version: Leaflet master as of yesterday (with #6855 in)

**Long tap on map:** works thanks to #6855, alert appears highlighting a contextmenu/longtap has been detected
**Long tap on marker:** user gets screen to save/share Marker icon, after closing it the alert appears

Expected behavior: when long tap on Marker OS menu to save image should not appear

`var map = L.map('map', { center: [44.22253, 11.77083], zoom: 4, maxZoom: 18, layers: [ L.tileLayer('https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png') ], attributionControl: false, tap: true});
map.on('contextmenu', function(e) {alert("MAP: contextmenu/longtap detected"); });
var m = L.marker([44.22253, 11.77083]);
m.on('contextmenu', function(e) {alert("MARKER***: contextmenu/longtap detected"); } );
map.addLayer(m);`

**Comment by filcab** [comment]
(It's probably more appropriate to post here, instead of on the closed issue)

Hi @eliboni,
It seems you need to use -webkit-touch-callout: none; (or similar... I'm bad at this 😅) on the elements. Unsure on which, though. Adding * { -webkit-touch-callout: none; } worked on my test (contextmenu event triggered on a marker, using an iPhone).
Adjust as you need.

Maybe leaflet can have a class with that CSS for when `contextmenu` events are listened on by markers? Otherwise, it might best to fix on the page using leaflet?

**Comment by eliboni** [comment]
@filcab many thanks. It worked by adding this to just the leaflet-marker-icon class:

`.leaflet-marker-icon { -webkit-touch-callout: none; }`

Test URL: https://www.mappite.org/tap/testTap2.html

I agree on your suggestion. I wander why this was not needed in the past, is this really linked to #6817?

**Comment by filcab** [comment]
I think iOS13 might have changed something.
But I don't think this is 6817, as that bug was about events not triggering at all due to PointerEvents being available.
These aren't triggering due to webkit features for long tap, which have to be disabled.

**Comment by johnd0e** [comment]
I see similar issue on Android, with Firefox Preview 4.0.0

**Comment by eliboni** [comment]
Any news on this one? Also with 1.8.0, when long tapping on a marker (iOS 15.5), the event is properly handled by leaflet but the marker then gets magnified with teh option to close it. I'm wondering if this is connected with [@1926](https://github.com/lichess-org/lichobile/issues/1926) , anyway  "-webkit-user-select: none" does not seem to have solved the issue.

https://imgur.com/J8DE8bQ.png

**Comment by Falke-Design** [comment]
@eliboni does it also happens when you make a long click on the zoom control and while making a long click on the shadow of the marker?

**Comment by eliboni** [comment]
When clicking on shadow the behaviour is different, the "+" of the zoom
control gets selected! The same happens when long tap anywhere else on the
map

**Comment by Malvoz** [comment]
Re https://github.com/Leaflet/Leaflet/issues/6865#issuecomment-1133030697:

> added to default leaflet.css
> 
> `.leaflet-marker-icon { -webkit-touch-callout: none; }`

Sure, but that would only fix the issue in Safari.

Re https://github.com/Leaflet/Leaflet/issues/6865#issuecomment-1133116467:

> When clicking on shadow the behaviour is different, the "+" of the zoom
> control gets selected!

That's about text selection, separate issue: https://github.com/Leaflet/Leaflet/issues/6872.

(However, even the `<a href="#">` controls (e.g. zoom controls) can prompt the URL "touch callout", that issue would be fixed by either https://github.com/Leaflet/Leaflet/issues/7821 or https://github.com/Leaflet/Leaflet/issues/7368.)

**Comment by eliboni** [comment]
> Re [#6865 (comment)](https://github.com/Leaflet/Leaflet/issues/6865#issuecomment-1133030697):
> 
> > added to default leaflet.css
> > `.leaflet-marker-icon { -webkit-touch-callout: none; }`
> 
> Sure, but that would only fix the issue in Safari.
> 


Well, the issue is *only* with Safari

**Comment by Malvoz** [comment]
@eliboni No, see https://github.com/Leaflet/Leaflet/issues/6865#issuecomment-599536458. I can also reproduce on Android.

**Comment by eliboni** [comment]
EDIT: corrected jsfiddle.net link above, the issue does not appear with older leaflet versions (1.0 for example).
